scripts/microbe_stage/microbe_operations.as

```diff
@@ -912,15 +914,11 @@ ObjectID spawnBacteria(CellStageWorld@ world, Float3 pos, const string &in speci
     node.Node.setPosition(pos);
 
     // Bacteria get scaled to half size
-    // TODO: wow, this is a big hack and no way guarantees that
-    // the physics size matches the rendered size
     node.Scale = Float3(0.5, 0.5, 0.5);
     node.Marked = true;
-    // This call is also not the cheapest. So would be much better
-    // if the physics generation actually did the right then when
-    // species.isBacteria is true
-    physics.ChangeShape(world.GetPhysicalWorld(),
-        world.GetPhysicalWorld().CreateSphere(HEX_SIZE/2.0f));
+
+    // ! We add physics to fit with render using restoreOrganelleLayout function
+    Species::restoreOrganelleLayout(world, microbeEntity, microbeComponent, species);
```

Comment:
Calling `Species::restoreOrganelleLayout` from the *spawn* function is not good. That function is supposed to help *reset* cell state. You should change `_createMicrobeEntity` to do the right thing.

Reply:
Do you think just move restoreOrganelleLayout in _createMicrobeEntity entity should be enough or it's better find a way to do the same thing that function do just for what we need?

Reply:
No. The point is that _createMicrobeEntity calls Species::applyTemplate. And `restoreOrganelleLayout` is a variant of applyTemplate that works for resetting organelles. It makes no sense to want to call restoreOrganelleLayout when *spawning* a cell.

Reply:
Actually i just notice that  Species::applyTemplate call already restoreOrganelleLayout, so nothing as to be implemented more, restoreOrganelle go trough addOrganelle in microbeOperations and then in onAddOrganelle of organelle.as that with other modifications do right things if species is bacteria or not